waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -80,32 +81,40 @@ public PrefixBasedDatabaseMappingService(
     init(initialMetastores);
   }
 
-  private void init(List<AbstractMetaStore> federatedMetaStores) {
+  private void init(List<AbstractMetaStore> abstractMetaStores) {
     mappingsByPrefix = Collections.synchronizedMap(new LinkedHashMap<>());
     mappedDbByPrefix = new ConcurrentHashMap<>();
-    for (AbstractMetaStore federatedMetaStore : federatedMetaStores) {
-      add(federatedMetaStore);
+    for (AbstractMetaStore abstractMetaStore : abstractMetaStores) {
+      add(abstractMetaStore);
     }
   }
 
-  private void add(AbstractMetaStore federatedMetaStore) {
-    MetaStoreMapping metaStoreMapping = metaStoreMappingFactory.newInstance(federatedMetaStore);
+  private void add(AbstractMetaStore metaStore) {
+    MetaStoreMapping metaStoreMapping = metaStoreMappingFactory.newInstance(metaStore);
 
-    if (federatedMetaStore.getFederationType() == PRIMARY) {
+    if (metaStore.getFederationType() == PRIMARY) {
       primaryDatabaseMapping = createDatabaseMapping(metaStoreMapping);
       mappingsByPrefix.put(metaStoreMapping.getDatabasePrefix(), primaryDatabaseMapping);
+      primaryShouldMatchAllDatabases =
+          !metaStore.shouldHaveNoMappedDatabases() && metaStore.getMappedDatabases().isEmpty();
     } else {
       mappingsByPrefix.put(metaStoreMapping.getDatabasePrefix(), createDatabaseMapping(metaStoreMapping));
-      Whitelist mappedDbWhitelist;
-      if (FederatedMetaStore.class.isAssignableFrom(federatedMetaStore.getClass())) {
-        mappedDbWhitelist = getWhitelistedDatabases((FederatedMetaStore) federatedMetaStore);
-        mappedDbByPrefix.put(metaStoreMapping.getDatabasePrefix(), mappedDbWhitelist);
-      }
     }
+
+    Whitelist mappedDbWhitelist = getWhitelistedDatabases(metaStore);
+    mappedDbByPrefix.put(metaStoreMapping.getDatabasePrefix(), mappedDbWhitelist);
   }
 
-  private Whitelist getWhitelistedDatabases(FederatedMetaStore federatedMetaStore) {
-    return new Whitelist(federatedMetaStore.getMappedDatabases());
+  private Whitelist getWhitelistedDatabases(AbstractMetaStore metaStore) {
+    List<String> databasesPatternToMap;
+
+    // check if metastore is supposed to have all databases matched or not
+    if (!metaStore.shouldHaveNoMappedDatabases() && metaStore.getMappedDatabases().isEmpty()) {
```

Comment:
Do you need to check both these things here ? I think `shouldHaveNoMappedDatabases()` is supposed to handle both null and empty usecase.

Reply:
You're right, the first check should be enough. Thanks :)